File: corpus/native/zig/hello.zig
const std = @import("std");

fn fib(n: u64) u64 {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

fn greet(writer: anytype, name: []const u8) !void {
    try writer.print("hello, {s}!\n", .{name});
}

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try greet(stdout, "disrobe");
    try stdout.print("fib(10) = {d}\n", .{fib(10)});
}

File: corpus/native/nim/hello.nim
proc fib(n: int): int =
  if n < 2: n
  else: fib(n - 1) + fib(n - 2)

proc greet(name: string): string =
  "hello, " & name & "!"

when isMainModule:
  echo greet("disrobe")
  echo "fib(10) = ", fib(10)
